### PR TITLE
network-ng: Add infiniband connection

### PR DESCRIPTION
**Pull request by teclator**
- https://trello.com/c/DT4tgPgw/1145-3-add-missing-connectionconfig-classes

### CODE REVIEW

File: src/lib/y2network/connection_config/infiniband.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+require "y2network/connection_config/base"
+
+module Y2Network
+  module ConnectionConfig
+    # Configuration for infiniband connections
+    #
+    # @see https://www.kernel.org/doc/Documentation/infiniband/ipoib.txt
+    class Infiniband < Base
+      # @return [String] transport mode ("datagram" or "connected")
```
Comment (imobachgs) [inline src/lib/y2network/connection_config/infiniband.rb]: Why not an enum? (`BootProtocol`-like).

Comment (jreidinger) [inline src/lib/y2network/connection_config/infiniband.rb]: what means empty value? Use default?

File: src/lib/y2network/ipoib_mode.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+module Y2Network
+  # This class represents the supported IPoIB transport modes.
+  class IpoibMode
+    class << self
+      # Returns all the existing modes
+      #
+      # @return [Array<IpoibMode>]
+      def all
+        @all ||= IpoibMode.constants
```
Comment (imobachgs) [inline src/lib/y2network/ipoib_mode.rb]: Not for this PR, but we might think about putting this code in a common place (a mixin or something like that).

File: test/y2network/sysconfig/connection_config_writers/infiniband_test.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    ]
+  end
+
+  let(:conn) do
+    instance_double(
+      Y2Network::ConnectionConfig::Infiniband,
+      name:        "ib0",
+      interface:   "ib0",
+      description: "",
+      ipoib_mode:  "datagram",
```
Comment (imobachgs) [inline test/y2network/sysconfig/connection_config_writers/infiniband_test.rb]: This should be a IpoibMode instance, right?

Comment (teclator) [reply test/y2network/sysconfig/connection_config_writers/infiniband_test.rb]: Right

**Pull request merged**